src/CommonDBTM.php

```diff
@@ -5498,6 +5498,11 @@ public function addFiles(array $input, $options = [])
                 $is_recursive = $input['is_recursive'];
             } else if (isset($input['_job']->fields['is_recursive'])) {
                 $is_recursive = $input['_job']->fields['is_recursive'];
+            } else if ($this instanceof CommonDBVisible) {
+                // CommonDBVisible visibility restriction is unpredictable as
+                // it may change over time, and can be related to dynamic profiles assignation.
+                // Related documents have to be available on all entities.
+                $is_recursive = 1;
             }
```

Comment:
I think we should also force `entities_id` to `0` when `$this instanceof CommonDBVisible`.
```suggestion
            }

            if ($this instanceof CommonDBVisible) {
                // CommonDBVisible visibility restriction is unpredictable as
                // it may change over time, and can be related to dynamic profiles assignation.
                // Related documents have to be available on all entities.
                $entities_id  = 0;
                $is_recursive = 1;
            }
```

Reply:
@orthagh , do you agree with this?